test/common/router/router_test.cc

```diff
@@ -868,6 +873,154 @@ TEST_F(RouterTest, EnvoyUpstreamServiceTime) {
   EXPECT_TRUE(verifyHostUpstreamStats(1, 0));
 }
 
+// Validate that x-envoy-attempt-count is added when option is true.
```

Comment:
note that we should have unit tests like this for the request path. I created https://github.com/envoyproxy/envoy/issues/10366. I can do this in a subsequent PR to prevent adding more lines to this PR.